ansible_base/rbac/caching.py

```diff
@@ -0,0 +1,214 @@
+import logging
+from collections import defaultdict
+from typing import Optional
+from uuid import UUID
+
+from ansible_base.rbac.models import ObjectRole, RoleDefinition, RoleEvaluation, RoleEvaluationUUID
+from ansible_base.rbac.permission_registry import permission_registry
+from ansible_base.rbac.prefetch import TypesPrefetch
+
+logger = logging.getLogger('ansible_base.rbac.caching')
+
+
+'''
```

Comment:
Style: for whatever weird historical reasons the AWX codebase is littered with these single-triples (as well as having double-triples), but for this new codebase we should IMHO standardize and do all double-triples.